{% extends "guide_base.html" %}

{% block guide_content %}
<div style="padding: 20px 30px 20px 30px">
<h1>16S Visualization Pipeline</h1>
<nav id="TOC">
<ul>
<li><a href="#ampvis2-and-plotly">ampvis2 and Plotly</a></li>
<li><a href="#plots">Plots</a></li>
<li><a href="#output-files">Output Files</a></li>
<li><a href="#tools-and-references">Tools and References</a></li>
</ul>
</nav>
<style>
code > span.kw { color: #152166; font-weight: bold; } /* Keyword^ */
code > span.dt { color: #152166; } /* DataType^ */
code > span.fl { color: #099; } /* Float^ */
code > span.st { color: #d14; } /* String^ */
code > span.co { color: #8f5902; font-style: italic; } /* Comment */
code > span.ot { color: #990073; } /* Other^ */
code > span.op { color: #006666; font-weight: bold; } /* Operator^ */
</style>
<h2 id="ampvis2-and-plotly">ampvis2 and Plotly</h2>
<p>Nephele uses the <a href="https://madsalbertsen.github.io/ampvis2/" target="_blank" rel="noopener noreferrer">ampvis2 R package</a> v2.3.2 for statistical computation. We also make use of the <a href="https://plot.ly/r/" target="_blank" rel="noopener noreferrer">Plotly R interface</a> for the <a href="https://plot.ly" target="_blank" rel="noopener noreferrer">plotly.js</a> charting library. The plots can be minimally edited interactively. However, they also have a link to export the data to the <a href="https://plot.ly/online-chart-maker/" target="_blank" rel="noopener noreferrer">Plotly Chart Studio</a> which allows for <a href="https://help.plot.ly/tutorials/" target="_blank" rel="noopener noreferrer">making a variety of charts</a>. We have a tutorial for <a href="{{ url_for('show_tutorials') }}">making simple edits to the plots here</a>.</p>
<h2 id="plots">Plots</h2>
<h3 id="rarefaction-curve">Rarefaction Curve</h3>
<p>The rarefaction curves are made with the <a href="https://madsalbertsen.github.io/ampvis2/reference/amp_rarecurve.html" target="_blank" rel="noopener noreferrer">amp_rarecurve</a> function. The table is written to <em>rarecurve.txt</em> and the plot to <em>rarecurve.html</em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">rarecurve &lt;-<span class="st"> </span><span class="kw">amp_rarecurve</span>(amp, <span class="dt">color_by =</span> <span class="st">"TreatmentGroup"</span>)</code></pre></div>
<h3 id="pcoa">PCoA</h3>
<p>Principal coordinates analysis using binomial and Bray-Curtis distances is carried out using <a href="https://madsalbertsen.github.io/ampvis2/reference/amp_ordinate.html" target="_blank" rel="noopener noreferrer">amp_ordinate</a>. Samples which fall below the specified sampling depth are removed from the OTU table using <a href="https://madsalbertsen.github.io/ampvis2/reference/amp_subset_samples.html" target="_blank" rel="noopener noreferrer">amp_subset_samples</a> before the distance measures are computed. The names of samples that are removed are output to <em>samples_being_ignored.txt</em>.</p>
<p>The binomial distance is able to handle varying sample sizes. However, for the Bray-Curtis distance, the OTU table is rarefied to the sampling depth using <a href="https://www.rdocumentation.org/packages/vegan/versions/2.4-2/topics/rarefy" target="_blank" rel="noopener noreferrer">rrarefy</a> from the vegan R package. At least 3 samples are needed in the OTU table for the plot to be generated. The tables are written to <em>pcoa_binomial.txt</em> and <em>pcoa_bray.txt</em> and the plots to <em>pcoa_binomial.html</em> and <em>pcoa_bray.html</em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">amp &lt;-<span class="st"> </span><span class="kw">amp_subset_samples</span>(amp, <span class="dt">minreads =</span> sampdepth)
pcoa_binomial &lt;-<span class="st"> </span><span class="kw">amp_ordinate</span>(amp, <span class="dt">filter_species =</span> <span class="fl">0.01</span>, <span class="dt">type =</span> <span class="st">"PCOA"</span>, 
    <span class="dt">distmeasure =</span> <span class="st">"binomial"</span>, <span class="dt">sample_color_by =</span> <span class="st">"TreatmentGroup"</span>, 
    <span class="dt">detailed_output =</span> <span class="ot">TRUE</span>, <span class="dt">transform =</span> <span class="st">"none"</span>)
otu &lt;-<span class="st"> </span><span class="kw">rrarefy</span>(<span class="kw">t</span>(amp<span class="op">$</span>abund), sampdepth)
amp<span class="op">$</span>abund &lt;-<span class="st"> </span><span class="kw">t</span>(otu)
pcoa_bray &lt;-<span class="st"> </span><span class="kw">amp_ordinate</span>(amp, <span class="dt">filter_species =</span> <span class="fl">0.01</span>, <span class="dt">type =</span> <span class="st">"PCOA"</span>, 
    <span class="dt">distmeasure =</span> <span class="st">"bray"</span>, <span class="dt">sample_color_by =</span> <span class="st">"TreatmentGroup"</span>, 
    <span class="dt">detailed_output =</span> <span class="ot">TRUE</span>, <span class="dt">transform =</span> <span class="st">"none"</span>)</code></pre></div>
<h3 id="alpha-diversity">Alpha diversity</h3>
<p>The Shannon diversity and Chao species richness are computed using <a href="https://madsalbertsen.github.io/ampvis2/reference/amp_alphadiv.html" target="_blank" rel="noopener noreferrer">amp_alphadiv</a>. Samples which fall below the specified sampling depth are removed, and the OTU table is rarefied to the sampling depth before the diversity computation. The table is written to <em>alphadiv.txt</em>, and boxplots are output to <em>alphadiv.html</em>. At least 3 samples are needed to produce these plots.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">alphadiv &lt;-<span class="st"> </span><span class="kw">amp_alphadiv</span>(amp, <span class="dt">measure =</span> <span class="st">"shannon"</span>, <span class="dt">richness =</span> <span class="ot">TRUE</span>, <span class="dt">rarefy =</span> sampdepth)</code></pre></div>
<h3 id="heatmap">Heatmap</h3>
<p>The interactive heatmap is implemented using the <a href="https://github.com/cmap/morpheus.R" target="_blank" rel="noopener noreferrer">morpheus R API</a> developed at the Broad Institute. <a href="https://software.broadinstitute.org/morpheus/documentation.html" target="_blank" rel="noopener noreferrer">Documentation</a> for how to use the heatmap can be found on the <a href="https://software.broadinstitute.org/morpheus/" target="_blank" rel="noopener noreferrer">morpheus website</a>.</p>
<p>The OTU table is first normalized to represent the relative abundances using amp_subset_samples before the heatmap is made with morpheus. In order for the heatmap to be generated, the OTU table must be at least 2x2. The heatmap is made from the raw OTU table (<em>seq_heatmap.html</em>).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">amptax &lt;-<span class="st"> </span><span class="kw">amp_subset_samples</span>(amp, <span class="dt">normalise =</span> <span class="ot">TRUE</span>)
heatmap &lt;-<span class="st"> </span><span class="kw">morpheus</span>(amp<span class="op">$</span>abund, <span class="dt">columns =</span> columns, <span class="dt">rows =</span> rows, 
    <span class="dt">columnColorModel =</span> <span class="kw">list</span>(<span class="dt">type =</span> <span class="kw">as.list</span>(colors)), 
    <span class="dt">colorScheme =</span> <span class="kw">list</span>(<span class="dt">scalingMode =</span> <span class="st">"fixed"</span>, <span class="dt">stepped =</span> <span class="ot">FALSE</span>), 
    <span class="dt">columnAnnotations =</span> amptax<span class="op">$</span>metadata, <span class="dt">rowAnnotations =</span> amptax<span class="op">$</span>tax)</code></pre></div>
<h2 id="output-files">Output Files</h2>
<p>Complete descriptions of the output files can be found in the <a href="#plots">Plots section above</a>. To learn how to edit the plots, see the <a href="{{ url_for('show_tutorials') }}">visualization tutorial</a>.</p>
<ul>
<li><em>rarecurve.html</em>: rarefaction curve plot</li>
<li><em>rarecurve.txt</em>: tabular data used to make the rarefaction curve plot</li>
<li><em>samples_being_ignored.txt</em>: list of samples removed from the analysis</li>
<li><em>pcoa_*.html</em>: PCoA plots</li>
<li><em>pcoa_*.txt</em>: tabular data used to make the PCoA plots</li>
<li><em>alphadiv.html</em>: alpha diversity boxplots</li>
<li><em>alphadiv.txt</em>: tabular data used to make the alpha diversity plots</li>
<li><em>seq_heatmap.html</em>: heatmap of OTU/sequence variant abundances</li>
</ul>
<h2 id="tools-and-references">Tools and References</h2>
<p>
M A, SM K, AS Z, RH K and PH N (2015). "Back to Basics - The Influence of DNA Extraction and Primer Choice on Phylogenetic Analysis of Activated Sludge Communities." <em>PLoS ONE</em>, <b>10</b>(7), pp. e0132783. <a href="http://dx.plos.org/10.1371/journal.pone.0132783" target="_blank" rel="noopener noreferrer">http://dx.plos.org/10.1371/journal.pone.0132783</a>.
</p>
<p>
Gould J (2018). <em>morpheus: Interactive heat maps using 'morpheus.js' and 'htmlwidgets'</em>. R package version 0.1.1.1, <a href="https://github.com/cmap/morpheus.R" target="_blank" rel="noopener noreferrer">https://github.com/cmap/morpheus.R</a>.
</p>
<p>
Sievert C, Parmer C, Hocking T, Chamberlain S, Ram K, Corvellec M and Despouy P (2017). <em>plotly: Create Interactive Web Graphics via 'plotly.js'</em>. R package version 4.7.1, <a href="https://CRAN.R-project.org/package=plotly" target="_blank" rel="noopener noreferrer">https://CRAN.R-project.org/package=plotly</a>.
</p>
<p>
McMurdie PJ and Paulson JN (2016). <em>biomformat: An interface package for the BIOM file format</em>. <a href="https://github.com/joey711/biomformat/" target="_blank" rel="noopener noreferrer">https://github.com/joey711/biomformat/</a>.
</p>
</div>
{% endblock %}
