<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta name="viewport" content="width=device-width, initial-scale=1">

<link href="data:text/css;charset=utf-8,%0A%40font%2Dface%20%7B%0Afont%2Dfamily%3A%20octicons%2Dlink%3B%0Asrc%3A%20url%28data%3Afont%2Fwoff%3Bcharset%3Dutf%2D8%3Bbase64%2Cd09GRgABAAAAAAZwABAAAAAACFQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABEU0lHAAAGaAAAAAgAAAAIAAAAAUdTVUIAAAZcAAAACgAAAAoAAQAAT1MvMgAAAyQAAABJAAAAYFYEU3RjbWFwAAADcAAAAEUAAACAAJThvmN2dCAAAATkAAAABAAAAAQAAAAAZnBnbQAAA7gAAACyAAABCUM%2B8IhnYXNwAAAGTAAAABAAAAAQABoAI2dseWYAAAFsAAABPAAAAZwcEq9taGVhZAAAAsgAAAA0AAAANgh4a91oaGVhAAADCAAAABoAAAAkCA8DRGhtdHgAAAL8AAAADAAAAAwGAACfbG9jYQAAAsAAAAAIAAAACABiATBtYXhwAAACqAAAABgAAAAgAA8ASm5hbWUAAAToAAABQgAAAlXu73sOcG9zdAAABiwAAAAeAAAAME3QpOBwcmVwAAAEbAAAAHYAAAB%2FaFGpk3jaTY6xa8JAGMW%2FO62BDi0tJLYQincXEypYIiGJjSgHniQ6umTsUEyLm5BV6NDBP8Tpts6F0v%2Bk%2F0an2i%2BitHDw3v2%2B9%2BDBKTzsJNnWJNTgHEy4BgG3EMI9DCEDOGEXzDADU5hBKMIgNPZqoD3SilVaXZCER3%2FI7AtxEJLtzzuZfI%2BVVkprxTlXShWKb3TBecG11rwoNlmmn1P2WYcJczl32etSpKnziC7lQyWe1smVPy%2FLt7Kc%2B0vWY%2FgAgIIEqAN9we0pwKXreiMasxvabDQMM4riO%2BqxM2ogwDGOZTXxwxDiycQIcoYFBLj5K3EIaSctAq2kTYiw%2Bymhce7vwM9jSqO8JyVd5RH9gyTt2%2BJ%2FyUmYlIR0s04n6%2B7Vm1ozezUeLEaUjhaDSuXHwVRgvLJn1tQ7xiuVv%2FocTRF42mNgZGBgYGbwZOBiAAFGJBIMAAizAFoAAABiAGIAznjaY2BkYGAA4in8zwXi%2BW2%2BMjCzMIDApSwvXzC97Z4Ig8N%2FBxYGZgcgl52BCSQKAA3jCV8CAABfAAAAAAQAAEB42mNgZGBg4f3vACQZQABIMjKgAmYAKEgBXgAAeNpjYGY6wTiBgZWBg2kmUxoDA4MPhGZMYzBi1AHygVLYQUCaawqDA4PChxhmh%2F8ODDEsvAwHgMKMIDnGL0x7gJQCAwMAJd4MFwAAAHjaY2BgYGaA4DAGRgYQkAHyGMF8NgYrIM3JIAGVYYDT%2BAEjAwuDFpBmA9KMDEwMCh9i%2Fv8H8sH0%2F4dQc1iAmAkALaUKLgAAAHjaTY9LDsIgEIbtgqHUPpDi3gPoBVyRTmTddOmqTXThEXqrob2gQ1FjwpDvfwCBdmdXC5AVKFu3e5MfNFJ29KTQT48Ob9%2FlqYwOGZxeUelN2U2R6%2BcArgtCJpauW7UQBqnFkUsjAY%2FkOU1cP%2BDAgvxwn1chZDwUbd6CFimGXwzwF6tPbFIcjEl%2BvvmM%2FbyA48e6tWrKArm4ZJlCbdsrxksL1AwWn%2FyBSJKpYbq8AXaaTb8AAHja28jAwOC00ZrBeQNDQOWO%2F%2FsdBBgYGRiYWYAEELEwMTE4uzo5Zzo5b2BxdnFOcALxNjA6b2ByTswC8jYwg0VlNuoCTWAMqNzMzsoK1rEhNqByEyerg5PMJlYuVueETKcd%2F89uBpnpvIEVomeHLoMsAAe1Id4AAAAAAAB42oWQT07CQBTGv0JBhagk7HQzKxca2sJCE1hDt4QF%2B9JOS0nbaaYDCQfwCJ7Au3AHj%2BLO13FMmm6cl7785vven0kBjHCBhfpYuNa5Ph1c0e2Xu3jEvWG7UdPDLZ4N92nOm%2BEBXuAbHmIMSRMs%2B4aUEd4Nd3CHD8NdvOLTsA2GL8M9PODbcL%2BhD7C1xoaHeLJSEao0FEW14ckxC%2BTU8TxvsY6X0eLPmRhry2WVioLpkrbp84LLQPGI7c6sOiUzpWIWS5GzlSgUzzLBSikOPFTOXqly7rqx0Z1Q5BAIoZBSFihQYQOOBEdkCOgXTOHA07HAGjGWiIjaPZNW13%2F%2Blm6S9FT7rLHFJ6fQbkATOG1j2OFMucKJJsxIVfQORl%2B9Jyda6Sl1dUYhSCm1dyClfoeDve4qMYdLEbfqHf3O%2FAdDumsjAAB42mNgYoAAZQYjBmyAGYQZmdhL8zLdDEydARfoAqIAAAABAAMABwAKABMAB%2F%2F%2FAA8AAQAAAAAAAAAAAAAAAAABAAAAAA%3D%3D%29%20format%28%27woff%27%29%3B%0A%7D%0Abody%20%7B%0A%2Dwebkit%2Dtext%2Dsize%2Dadjust%3A%20100%25%3B%0Atext%2Dsize%2Dadjust%3A%20100%25%3B%0Acolor%3A%20%23333%3B%0Afont%2Dfamily%3A%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20%22Segoe%20UI%22%2C%20Arial%2C%20freesans%2C%20sans%2Dserif%2C%20%22Apple%20Color%20Emoji%22%2C%20%22Segoe%20UI%20Emoji%22%2C%20%22Segoe%20UI%20Symbol%22%3B%0Afont%2Dsize%3A%2016px%3B%0Aline%2Dheight%3A%201%2E6%3B%0Aword%2Dwrap%3A%20break%2Dword%3B%0A%7D%0Aa%20%7B%0Abackground%2Dcolor%3A%20transparent%3B%0A%7D%0Aa%3Aactive%2C%0Aa%3Ahover%20%7B%0Aoutline%3A%200%3B%0A%7D%0Astrong%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Ah1%20%7B%0Afont%2Dsize%3A%202em%3B%0Amargin%3A%200%2E67em%200%3B%0A%7D%0Aimg%20%7B%0Aborder%3A%200%3B%0A%7D%0Ahr%20%7B%0Abox%2Dsizing%3A%20content%2Dbox%3B%0Aheight%3A%200%3B%0A%7D%0Apre%20%7B%0Aoverflow%3A%20auto%3B%0A%7D%0Acode%2C%0Akbd%2C%0Apre%20%7B%0Afont%2Dfamily%3A%20monospace%2C%20monospace%3B%0Afont%2Dsize%3A%201em%3B%0A%7D%0Ainput%20%7B%0Acolor%3A%20inherit%3B%0Afont%3A%20inherit%3B%0Amargin%3A%200%3B%0A%7D%0Ahtml%20input%5Bdisabled%5D%20%7B%0Acursor%3A%20default%3B%0A%7D%0Ainput%20%7B%0Aline%2Dheight%3A%20normal%3B%0A%7D%0Ainput%5Btype%3D%22checkbox%22%5D%20%7B%0Abox%2Dsizing%3A%20border%2Dbox%3B%0Apadding%3A%200%3B%0A%7D%0Atable%20%7B%0Aborder%2Dcollapse%3A%20collapse%3B%0Aborder%2Dspacing%3A%200%3B%0A%7D%0Atd%2C%0Ath%20%7B%0Apadding%3A%200%3B%0A%7D%0A%2A%20%7B%0Abox%2Dsizing%3A%20border%2Dbox%3B%0A%7D%0Ainput%20%7B%0Afont%3A%2013px%20%2F%201%2E4%20Helvetica%2C%20arial%2C%20nimbussansl%2C%20liberationsans%2C%20freesans%2C%20clean%2C%20sans%2Dserif%2C%20%22Apple%20Color%20Emoji%22%2C%20%22Segoe%20UI%20Emoji%22%2C%20%22Segoe%20UI%20Symbol%22%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%234078c0%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%2C%0Aa%3Aactive%20%7B%0Atext%2Ddecoration%3A%20underline%3B%0A%7D%0Ahr%20%7B%0Aheight%3A%200%3B%0Amargin%3A%2015px%200%3B%0Aoverflow%3A%20hidden%3B%0Abackground%3A%20transparent%3B%0Aborder%3A%200%3B%0Aborder%2Dbottom%3A%201px%20solid%20%23ddd%3B%0A%7D%0Ahr%3Abefore%20%7B%0Adisplay%3A%20table%3B%0Acontent%3A%20%22%22%3B%0A%7D%0Ahr%3Aafter%20%7B%0Adisplay%3A%20table%3B%0Aclear%3A%20both%3B%0Acontent%3A%20%22%22%3B%0A%7D%0Ah1%2C%0Ah2%2C%0Ah3%2C%0Ah4%2C%0Ah5%2C%0Ah6%20%7B%0Amargin%2Dtop%3A%2015px%3B%0Amargin%2Dbottom%3A%2015px%3B%0Aline%2Dheight%3A%201%2E1%3B%0A%7D%0Ah1%20%7B%0Afont%2Dsize%3A%2030px%3B%0A%7D%0Ah2%20%7B%0Afont%2Dsize%3A%2021px%3B%0A%7D%0Ah3%20%7B%0Afont%2Dsize%3A%2016px%3B%0A%7D%0Ah4%20%7B%0Afont%2Dsize%3A%2014px%3B%0A%7D%0Ah5%20%7B%0Afont%2Dsize%3A%2012px%3B%0A%7D%0Ah6%20%7B%0Afont%2Dsize%3A%2011px%3B%0A%7D%0Ablockquote%20%7B%0Amargin%3A%200%3B%0A%7D%0Aul%2C%0Aol%20%7B%0Apadding%3A%200%3B%0Amargin%2Dtop%3A%200%3B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Aol%20ol%2C%0Aul%20ol%20%7B%0Alist%2Dstyle%2Dtype%3A%20lower%2Droman%3B%0A%7D%0Aul%20ul%20ol%2C%0Aul%20ol%20ol%2C%0Aol%20ul%20ol%2C%0Aol%20ol%20ol%20%7B%0Alist%2Dstyle%2Dtype%3A%20lower%2Dalpha%3B%0A%7D%0Add%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20%22Liberation%20Mono%22%2C%20Menlo%2C%20Courier%2C%20monospace%3B%0Afont%2Dsize%3A%2012px%3B%0A%7D%0Apre%20%7B%0Amargin%2Dtop%3A%200%3B%0Amargin%2Dbottom%3A%200%3B%0Afont%3A%2012px%20Consolas%2C%20%22Liberation%20Mono%22%2C%20Menlo%2C%20Courier%2C%20monospace%3B%0A%7D%0A%2Eselect%3A%3A%2Dms%2Dexpand%20%7B%0Aopacity%3A%200%3B%0A%7D%0A%2Eocticon%20%7B%0Afont%3A%20normal%20normal%20normal%2016px%2F1%20octicons%2Dlink%3B%0Adisplay%3A%20inline%2Dblock%3B%0Atext%2Ddecoration%3A%20none%3B%0Atext%2Drendering%3A%20auto%3B%0A%2Dwebkit%2Dfont%2Dsmoothing%3A%20antialiased%3B%0A%2Dmoz%2Dosx%2Dfont%2Dsmoothing%3A%20grayscale%3B%0A%2Dwebkit%2Duser%2Dselect%3A%20none%3B%0A%2Dmoz%2Duser%2Dselect%3A%20none%3B%0A%2Dms%2Duser%2Dselect%3A%20none%3B%0Auser%2Dselect%3A%20none%3B%0A%7D%0A%2Eocticon%2Dlink%3Abefore%20%7B%0Acontent%3A%20%27%5Cf05c%27%3B%0A%7D%0A%2Emarkdown%2Dbody%3Abefore%20%7B%0Adisplay%3A%20table%3B%0Acontent%3A%20%22%22%3B%0A%7D%0A%2Emarkdown%2Dbody%3Aafter%20%7B%0Adisplay%3A%20table%3B%0Aclear%3A%20both%3B%0Acontent%3A%20%22%22%3B%0A%7D%0A%2Emarkdown%2Dbody%3E%2A%3Afirst%2Dchild%20%7B%0Amargin%2Dtop%3A%200%20%21important%3B%0A%7D%0A%2Emarkdown%2Dbody%3E%2A%3Alast%2Dchild%20%7B%0Amargin%2Dbottom%3A%200%20%21important%3B%0A%7D%0Aa%3Anot%28%5Bhref%5D%29%20%7B%0Acolor%3A%20inherit%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0A%2Eanchor%20%7B%0Adisplay%3A%20inline%2Dblock%3B%0Apadding%2Dright%3A%202px%3B%0Amargin%2Dleft%3A%20%2D18px%3B%0A%7D%0A%2Eanchor%3Afocus%20%7B%0Aoutline%3A%20none%3B%0A%7D%0Ah1%2C%0Ah2%2C%0Ah3%2C%0Ah4%2C%0Ah5%2C%0Ah6%20%7B%0Amargin%2Dtop%3A%201em%3B%0Amargin%2Dbottom%3A%2016px%3B%0Afont%2Dweight%3A%20bold%3B%0Aline%2Dheight%3A%201%2E4%3B%0A%7D%0Ah1%20%2Eocticon%2Dlink%2C%0Ah2%20%2Eocticon%2Dlink%2C%0Ah3%20%2Eocticon%2Dlink%2C%0Ah4%20%2Eocticon%2Dlink%2C%0Ah5%20%2Eocticon%2Dlink%2C%0Ah6%20%2Eocticon%2Dlink%20%7B%0Acolor%3A%20%23000%3B%0Avertical%2Dalign%3A%20middle%3B%0Avisibility%3A%20hidden%3B%0A%7D%0Ah1%3Ahover%20%2Eanchor%2C%0Ah2%3Ahover%20%2Eanchor%2C%0Ah3%3Ahover%20%2Eanchor%2C%0Ah4%3Ahover%20%2Eanchor%2C%0Ah5%3Ahover%20%2Eanchor%2C%0Ah6%3Ahover%20%2Eanchor%20%7B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Ah1%3Ahover%20%2Eanchor%20%2Eocticon%2Dlink%2C%0Ah2%3Ahover%20%2Eanchor%20%2Eocticon%2Dlink%2C%0Ah3%3Ahover%20%2Eanchor%20%2Eocticon%2Dlink%2C%0Ah4%3Ahover%20%2Eanchor%20%2Eocticon%2Dlink%2C%0Ah5%3Ahover%20%2Eanchor%20%2Eocticon%2Dlink%2C%0Ah6%3Ahover%20%2Eanchor%20%2Eocticon%2Dlink%20%7B%0Avisibility%3A%20visible%3B%0A%7D%0Ah1%20%7B%0Apadding%2Dbottom%3A%200%2E3em%3B%0Afont%2Dsize%3A%202%2E25em%3B%0Aline%2Dheight%3A%201%2E2%3B%0Aborder%2Dbottom%3A%201px%20solid%20%23eee%3B%0A%7D%0Ah1%20%2Eanchor%20%7B%0Aline%2Dheight%3A%201%3B%0A%7D%0Ah2%20%7B%0Apadding%2Dbottom%3A%200%2E3em%3B%0Afont%2Dsize%3A%201%2E75em%3B%0Aline%2Dheight%3A%201%2E225%3B%0Aborder%2Dbottom%3A%201px%20solid%20%23eee%3B%0A%7D%0Ah2%20%2Eanchor%20%7B%0Aline%2Dheight%3A%201%3B%0A%7D%0Ah3%20%7B%0Afont%2Dsize%3A%201%2E5em%3B%0Aline%2Dheight%3A%201%2E43%3B%0A%7D%0Ah3%20%2Eanchor%20%7B%0Aline%2Dheight%3A%201%2E2%3B%0A%7D%0Ah4%20%7B%0Afont%2Dsize%3A%201%2E25em%3B%0A%7D%0Ah4%20%2Eanchor%20%7B%0Aline%2Dheight%3A%201%2E2%3B%0A%7D%0Ah5%20%7B%0Afont%2Dsize%3A%201em%3B%0A%7D%0Ah5%20%2Eanchor%20%7B%0Aline%2Dheight%3A%201%2E1%3B%0A%7D%0Ah6%20%7B%0Afont%2Dsize%3A%201em%3B%0Acolor%3A%20%23777%3B%0A%7D%0Ah6%20%2Eanchor%20%7B%0Aline%2Dheight%3A%201%2E1%3B%0A%7D%0Ap%2C%0Ablockquote%2C%0Aul%2C%0Aol%2C%0Adl%2C%0Atable%2C%0Apre%20%7B%0Amargin%2Dtop%3A%200%3B%0Amargin%2Dbottom%3A%2016px%3B%0A%7D%0Ahr%20%7B%0Aheight%3A%204px%3B%0Apadding%3A%200%3B%0Amargin%3A%2016px%200%3B%0Abackground%2Dcolor%3A%20%23e7e7e7%3B%0Aborder%3A%200%20none%3B%0A%7D%0Aul%2C%0Aol%20%7B%0Apadding%2Dleft%3A%202em%3B%0A%7D%0Aul%20ul%2C%0Aul%20ol%2C%0Aol%20ol%2C%0Aol%20ul%20%7B%0Amargin%2Dtop%3A%200%3B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Ali%3Ep%20%7B%0Amargin%2Dtop%3A%2016px%3B%0A%7D%0Adl%20%7B%0Apadding%3A%200%3B%0A%7D%0Adl%20dt%20%7B%0Apadding%3A%200%3B%0Amargin%2Dtop%3A%2016px%3B%0Afont%2Dsize%3A%201em%3B%0Afont%2Dstyle%3A%20italic%3B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Adl%20dd%20%7B%0Apadding%3A%200%2016px%3B%0Amargin%2Dbottom%3A%2016px%3B%0A%7D%0Ablockquote%20%7B%0Apadding%3A%200%2015px%3B%0Acolor%3A%20%23777%3B%0Aborder%2Dleft%3A%204px%20solid%20%23ddd%3B%0A%7D%0Ablockquote%3E%3Afirst%2Dchild%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Ablockquote%3E%3Alast%2Dchild%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Atable%20%7B%0Adisplay%3A%20block%3B%0Awidth%3A%20100%25%3B%0Aoverflow%3A%20auto%3B%0Aword%2Dbreak%3A%20normal%3B%0Aword%2Dbreak%3A%20keep%2Dall%3B%0A%7D%0Atable%20th%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Atable%20th%2C%0Atable%20td%20%7B%0Apadding%3A%206px%2013px%3B%0Aborder%3A%201px%20solid%20%23ddd%3B%0A%7D%0Atable%20tr%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Aborder%2Dtop%3A%201px%20solid%20%23ccc%3B%0A%7D%0Atable%20tr%3Anth%2Dchild%282n%29%20%7B%0Abackground%2Dcolor%3A%20%23f8f8f8%3B%0A%7D%0Aimg%20%7B%0Amax%2Dwidth%3A%20100%25%3B%0Abox%2Dsizing%3A%20content%2Dbox%3B%0Abackground%2Dcolor%3A%20%23fff%3B%0A%7D%0Acode%20%7B%0Apadding%3A%200%3B%0Apadding%2Dtop%3A%200%2E2em%3B%0Apadding%2Dbottom%3A%200%2E2em%3B%0Amargin%3A%200%3B%0Afont%2Dsize%3A%2085%25%3B%0Abackground%2Dcolor%3A%20rgba%280%2C0%2C0%2C0%2E04%29%3B%0Aborder%2Dradius%3A%203px%3B%0A%7D%0Acode%3Abefore%2C%0Acode%3Aafter%20%7B%0Aletter%2Dspacing%3A%20%2D0%2E2em%3B%0Acontent%3A%20%22%5C00a0%22%3B%0A%7D%0Apre%3Ecode%20%7B%0Apadding%3A%200%3B%0Amargin%3A%200%3B%0Afont%2Dsize%3A%20100%25%3B%0Aword%2Dbreak%3A%20normal%3B%0Awhite%2Dspace%3A%20pre%3B%0Abackground%3A%20transparent%3B%0Aborder%3A%200%3B%0A%7D%0A%2Ehighlight%20%7B%0Amargin%2Dbottom%3A%2016px%3B%0A%7D%0A%2Ehighlight%20pre%2C%0Apre%20%7B%0Apadding%3A%2016px%3B%0Aoverflow%3A%20auto%3B%0Afont%2Dsize%3A%2085%25%3B%0Aline%2Dheight%3A%201%2E45%3B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0A%7D%0A%2Ehighlight%20pre%20%7B%0Amargin%2Dbottom%3A%200%3B%0Aword%2Dbreak%3A%20normal%3B%0A%7D%0Apre%20%7B%0Aword%2Dwrap%3A%20normal%3B%0A%7D%0Apre%20code%20%7B%0Adisplay%3A%20inline%3B%0Amax%2Dwidth%3A%20initial%3B%0Apadding%3A%200%3B%0Amargin%3A%200%3B%0Aoverflow%3A%20initial%3B%0Aline%2Dheight%3A%20inherit%3B%0Aword%2Dwrap%3A%20normal%3B%0Abackground%2Dcolor%3A%20transparent%3B%0Aborder%3A%200%3B%0A%7D%0Apre%20code%3Abefore%2C%0Apre%20code%3Aafter%20%7B%0Acontent%3A%20normal%3B%0A%7D%0Akbd%20%7B%0Adisplay%3A%20inline%2Dblock%3B%0Apadding%3A%203px%205px%3B%0Afont%2Dsize%3A%2011px%3B%0Aline%2Dheight%3A%2010px%3B%0Acolor%3A%20%23555%3B%0Avertical%2Dalign%3A%20middle%3B%0Abackground%2Dcolor%3A%20%23fcfcfc%3B%0Aborder%3A%20solid%201px%20%23ccc%3B%0Aborder%2Dbottom%2Dcolor%3A%20%23bbb%3B%0Aborder%2Dradius%3A%203px%3B%0Abox%2Dshadow%3A%20inset%200%20%2D1px%200%20%23bbb%3B%0A%7D%0A%2Epl%2Dc%20%7B%0Acolor%3A%20%23969896%3B%0A%7D%0A%2Epl%2Dc1%2C%0A%2Epl%2Ds%20%2Epl%2Dv%20%7B%0Acolor%3A%20%230086b3%3B%0A%7D%0A%2Epl%2De%2C%0A%2Epl%2Den%20%7B%0Acolor%3A%20%23795da3%3B%0A%7D%0A%2Epl%2Ds%20%2Epl%2Ds1%2C%0A%2Epl%2Dsmi%20%7B%0Acolor%3A%20%23333%3B%0A%7D%0A%2Epl%2Dent%20%7B%0Acolor%3A%20%2363a35c%3B%0A%7D%0A%2Epl%2Dk%20%7B%0Acolor%3A%20%23a71d5d%3B%0A%7D%0A%2Epl%2Dpds%2C%0A%2Epl%2Ds%2C%0A%2Epl%2Ds%20%2Epl%2Dpse%20%2Epl%2Ds1%2C%0A%2Epl%2Dsr%2C%0A%2Epl%2Dsr%20%2Epl%2Dcce%2C%0A%2Epl%2Dsr%20%2Epl%2Dsra%2C%0A%2Epl%2Dsr%20%2Epl%2Dsre%20%7B%0Acolor%3A%20%23183691%3B%0A%7D%0A%2Epl%2Dv%20%7B%0Acolor%3A%20%23ed6a43%3B%0A%7D%0A%2Epl%2Did%20%7B%0Acolor%3A%20%23b52a1d%3B%0A%7D%0A%2Epl%2Dii%20%7B%0Abackground%2Dcolor%3A%20%23b52a1d%3B%0Acolor%3A%20%23f8f8f8%3B%0A%7D%0A%2Epl%2Dsr%20%2Epl%2Dcce%20%7B%0Acolor%3A%20%2363a35c%3B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0A%2Epl%2Dml%20%7B%0Acolor%3A%20%23693a17%3B%0A%7D%0A%2Epl%2Dmh%2C%0A%2Epl%2Dmh%20%2Epl%2Den%2C%0A%2Epl%2Dms%20%7B%0Acolor%3A%20%231d3e81%3B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0A%2Epl%2Dmq%20%7B%0Acolor%3A%20%23008080%3B%0A%7D%0A%2Epl%2Dmi%20%7B%0Acolor%3A%20%23333%3B%0Afont%2Dstyle%3A%20italic%3B%0A%7D%0A%2Epl%2Dmb%20%7B%0Acolor%3A%20%23333%3B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0A%2Epl%2Dmd%20%7B%0Abackground%2Dcolor%3A%20%23ffecec%3B%0Acolor%3A%20%23bd2c00%3B%0A%7D%0A%2Epl%2Dmi1%20%7B%0Abackground%2Dcolor%3A%20%23eaffea%3B%0Acolor%3A%20%2355a532%3B%0A%7D%0A%2Epl%2Dmdr%20%7B%0Acolor%3A%20%23795da3%3B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0A%2Epl%2Dmo%20%7B%0Acolor%3A%20%231d3e81%3B%0A%7D%0Akbd%20%7B%0Adisplay%3A%20inline%2Dblock%3B%0Apadding%3A%203px%205px%3B%0Afont%3A%2011px%20Consolas%2C%20%22Liberation%20Mono%22%2C%20Menlo%2C%20Courier%2C%20monospace%3B%0Aline%2Dheight%3A%2010px%3B%0Acolor%3A%20%23555%3B%0Avertical%2Dalign%3A%20middle%3B%0Abackground%2Dcolor%3A%20%23fcfcfc%3B%0Aborder%3A%20solid%201px%20%23ccc%3B%0Aborder%2Dbottom%2Dcolor%3A%20%23bbb%3B%0Aborder%2Dradius%3A%203px%3B%0Abox%2Dshadow%3A%20inset%200%20%2D1px%200%20%23bbb%3B%0A%7D%0A%2Etask%2Dlist%2Ditem%20%7B%0Alist%2Dstyle%2Dtype%3A%20none%3B%0A%7D%0A%2Etask%2Dlist%2Ditem%2B%2Etask%2Dlist%2Ditem%20%7B%0Amargin%2Dtop%3A%203px%3B%0A%7D%0A%2Etask%2Dlist%2Ditem%20input%20%7B%0Amargin%3A%200%200%2E35em%200%2E25em%20%2D1%2E6em%3B%0Avertical%2Dalign%3A%20middle%3B%0A%7D%0A%3Achecked%2B%2Eradio%2Dlabel%20%7B%0Az%2Dindex%3A%201%3B%0Aposition%3A%20relative%3B%0Aborder%2Dcolor%3A%20%234078c0%3B%0A%7D%0A%2EsourceLine%20%7B%0Adisplay%3A%20inline%2Dblock%3B%0A%7D%0Acode%20%2Ekw%20%7B%20color%3A%20%23000000%3B%20%7D%0Acode%20%2Edt%20%7B%20color%3A%20%23ed6a43%3B%20%7D%0Acode%20%2Edv%20%7B%20color%3A%20%23009999%3B%20%7D%0Acode%20%2Ebn%20%7B%20color%3A%20%23009999%3B%20%7D%0Acode%20%2Efl%20%7B%20color%3A%20%23009999%3B%20%7D%0Acode%20%2Ech%20%7B%20color%3A%20%23009999%3B%20%7D%0Acode%20%2Est%20%7B%20color%3A%20%23183691%3B%20%7D%0Acode%20%2Eco%20%7B%20color%3A%20%23969896%3B%20%7D%0Acode%20%2Eot%20%7B%20color%3A%20%230086b3%3B%20%7D%0Acode%20%2Eal%20%7B%20color%3A%20%23a61717%3B%20%7D%0Acode%20%2Efu%20%7B%20color%3A%20%2363a35c%3B%20%7D%0Acode%20%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%0Acode%20%2Ewa%20%7B%20color%3A%20%23000000%3B%20%7D%0Acode%20%2Ecn%20%7B%20color%3A%20%23008080%3B%20%7D%0Acode%20%2Esc%20%7B%20color%3A%20%23008080%3B%20%7D%0Acode%20%2Evs%20%7B%20color%3A%20%23183691%3B%20%7D%0Acode%20%2Ess%20%7B%20color%3A%20%23183691%3B%20%7D%0Acode%20%2Eim%20%7B%20color%3A%20%23000000%3B%20%7D%0Acode%20%2Eva%20%7Bcolor%3A%20%23008080%3B%20%7D%0Acode%20%2Ecf%20%7B%20color%3A%20%23000000%3B%20%7D%0Acode%20%2Eop%20%7B%20color%3A%20%23000000%3B%20%7D%0Acode%20%2Ebu%20%7B%20color%3A%20%23000000%3B%20%7D%0Acode%20%2Eex%20%7B%20color%3A%20%23000000%3B%20%7D%0Acode%20%2Epp%20%7B%20color%3A%20%23999999%3B%20%7D%0Acode%20%2Eat%20%7B%20color%3A%20%23008080%3B%20%7D%0Acode%20%2Edo%20%7B%20color%3A%20%23969896%3B%20%7D%0Acode%20%2Ean%20%7B%20color%3A%20%23008080%3B%20%7D%0Acode%20%2Ecv%20%7B%20color%3A%20%23008080%3B%20%7D%0Acode%20%2Ein%20%7B%20color%3A%20%23008080%3B%20%7D%0A" rel="stylesheet">
<style>
body {
  box-sizing: border-box;
  min-width: 200px;
  max-width: 980px;
  margin: 0 auto;
  padding: 45px;
  padding-top: 0px;
}
</style>

</head>

<body>

<h1 id="datavis16s">datavis16s</h1>
<ul>
<li><a href="#description">DESCRIPTION</a></li>
<li><a href="#exported">Exported</a>
<ul>
<li><a href="#adivboxplot"><code>adivboxplot</code></a></li>
<li><a href="#allgraphs"><code>allgraphs</code></a></li>
<li><a href="#morphheatmap"><code>morphheatmap</code></a></li>
<li><a href="#pcoaplot"><code>pcoaplot</code></a></li>
<li><a href="#rarefactioncurve"><code>rarefactioncurve</code></a></li>
<li><a href="#readindata"><code>readindata</code></a></li>
<li><a href="#trygraphwrapper"><code>trygraphwrapper</code></a></li>
</ul></li>
<li><a href="#internal">Internal</a>
<ul>
<li><a href="#datavis16s-package"><code>datavis16s-package</code></a></li>
<li><a href="#filterlowabund"><code>filterlowabund</code></a></li>
<li><a href="#gridcode"><code>gridCode</code></a></li>
<li><a href="#highertax"><code>highertax</code></a></li>
<li><a href="#logoutput"><code>logoutput</code></a></li>
<li><a href="#plotlygrid"><code>plotlyGrid</code></a></li>
<li><a href="#print_ampvis2"><code>print_ampvis2</code></a></li>
<li><a href="#save_fillhtml"><code>save_fillhtml</code></a></li>
<li><a href="#shortnames"><code>shortnames</code></a></li>
<li><a href="#subsetamp"><code>subsetamp</code></a></li>
</ul></li>
</ul>
<!-- toc -->
<p>March 18, 2018</p>
<h1 id="description">DESCRIPTION</h1>
<pre><code>Package: datavis16s
Title: Graphs for Nephele 16S Pipelines
Version: 0.1.0
Authors@R: person(&quot;Poorani&quot;, &quot;Subramanian&quot;, email = &quot;poorani.subramanian@nih.gov&quot;, role = c(&quot;aut&quot;, &quot;cre&quot;))
Description: betterbetterplots!
Depends: R (&gt;= 3.4.1)
License: none
Encoding: UTF-8
LazyData: true
RoxygenNote: 6.0.1
Imports:
  ampvis2,
  biomformat,
  bpexploder,
  data.table,
  ggplot2,
  htmltools,
  htmlwidgets,
  jsonlite,
  morpheus,
  plotly,
  RColorBrewer,
  vegan
URL: https://github.niaid.nih.gov/bcbb/nephele2/tree/master/pipelines/datavis16s</code></pre>
<h1 id="exported">Exported</h1>
<h2 id="adivboxplot"><code>adivboxplot</code></h2>
<p>Alpha diversity boxplot</p>
<h3 id="description-1">Description</h3>
<p>Plots exploding boxplot of shannon diversity and Chao species richness. If sampling depth is NULL, rarefies OTU table to the minimum readcount of any sample. If this is low, then the plot will fail.</p>
<h3 id="usage">Usage</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">adivboxplot</span>(mapfile, datafile, outdir, <span class="dt">amp =</span> <span class="ot">NULL</span>, <span class="dt">sampdepth =</span> <span class="ot">NULL</span>,
  <span class="dt">colors =</span> <span class="ot">NULL</span>, ...)</code></pre></div>
<h3 id="arguments">Arguments</h3>
<table>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>mapfile</code></td>
<td>full path to map file</td>
</tr>
<tr class="even">
<td><code>datafile</code></td>
<td>full path to input OTU file</td>
</tr>
<tr class="odd">
<td><code>outdir</code></td>
<td>full path to output directory</td>
</tr>
<tr class="even">
<td><code>amp</code></td>
<td>ampvis2 object. may be specified instead of mapfile and datafile</td>
</tr>
<tr class="odd">
<td><code>sampdepth</code></td>
<td>sampling depth</td>
</tr>
<tr class="even">
<td><code>colors</code></td>
<td>colors to use for plots</td>
</tr>
<tr class="odd">
<td><code>...</code></td>
<td>other parameters to pass to <a href="#readindata">readindata</a></td>
</tr>
</tbody>
</table>
<h3 id="value">Value</h3>
<p>Save alpha diversity boxplots to outdir.</p>
<h3 id="source">Source</h3>
<p><a href="../R/graphs.R">graphs.R</a></p>
<h2 id="allgraphs"><code>allgraphs</code></h2>
<p>Pipeline function</p>
<h3 id="description-2">Description</h3>
<p>Make all 4 types of graphs</p>
<h3 id="usage-1">Usage</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">allgraphs</span>(mapfile, datafile, outdir, <span class="dt">sampdepth =</span> <span class="ot">NULL</span>, ...)</code></pre></div>
<h3 id="arguments-1">Arguments</h3>
<table>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>mapfile</code></td>
<td>full path to map file</td>
</tr>
<tr class="even">
<td><code>datafile</code></td>
<td>full path to input OTU file (biom or see <a href="#readindata">readindata</a> )</td>
</tr>
<tr class="odd">
<td><code>outdir</code></td>
<td>full path to output directory</td>
</tr>
<tr class="even">
<td><code>sampdepth</code></td>
<td>sampling depth</td>
</tr>
<tr class="odd">
<td><code>...</code></td>
<td>other parameters to pass to <a href="#readindata">readindata</a></td>
</tr>
</tbody>
</table>
<h3 id="value-1">Value</h3>
<p>graphs are saved to outdir</p>
<h3 id="source-1">Source</h3>
<p><a href="../R/graphs.R">graphs.R</a></p>
<h2 id="morphheatmap"><code>morphheatmap</code></h2>
<p>Morpheus heatmap</p>
<h3 id="description-3">Description</h3>
<p>Creates heatmaps using Morpheus R API <a href="https://software.broadinstitute.org/morpheus/" class="uri">https://software.broadinstitute.org/morpheus/</a> . The heatmaps are made using relative abundances.</p>
<h3 id="usage-2">Usage</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">morphheatmap</span>(mapfile, datafile, outdir, <span class="dt">amp =</span> <span class="ot">NULL</span>, <span class="dt">sampdepth =</span> <span class="ot">NULL</span>,
  <span class="dt">rarefy =</span> <span class="ot">FALSE</span>, <span class="dt">filter_level =</span> <span class="fl">0.1</span>, <span class="dt">taxlevel =</span> <span class="kw">c</span>(<span class="st">&quot;seq&quot;</span>),
  <span class="dt">colors =</span> <span class="ot">NULL</span>, ...)</code></pre></div>
<h3 id="arguments-2">Arguments</h3>
<table style="width:43%;">
<colgroup>
<col width="19%"></col>
<col width="23%"></col>
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>mapfile</code></td>
<td>full path to mapping file</td>
</tr>
<tr class="even">
<td><code>datafile</code></td>
<td>full path to input OTU file (biom or see <a href="#readindata">readindata</a> )</td>
</tr>
<tr class="odd">
<td><code>outdir</code></td>
<td>full path to output directory</td>
</tr>
<tr class="even">
<td><code>amp</code></td>
<td>(Optional) ampvis2 object. may be specified instead of mapfile and datafile</td>
</tr>
<tr class="odd">
<td><code>sampdepth</code></td>
<td>sampling depth</td>
</tr>
<tr class="even">
<td><code>rarefy</code></td>
<td>Logical. Rarefy the OTU table if sampdepth is specified.</td>
</tr>
<tr class="odd">
<td><code>filter_level</code></td>
<td>minimum abundance to show in the heatmap</td>
</tr>
<tr class="even">
<td><code>taxlevel</code></td>
<td>vector of taxonomic levels to graph. must be subset of c(&quot;Kingdom&quot;, &quot;Phylum&quot;, &quot;Class&quot;, &quot;Order&quot;, &quot;Family&quot;, &quot;Genus&quot;, &quot;Species&quot;, &quot;seq&quot;). See Details.</td>
</tr>
<tr class="odd">
<td><code>colors</code></td>
<td>(Optional) color vector - length equal to number of TreatmentGroups in mapfile</td>
</tr>
<tr class="even">
<td><code>...</code></td>
<td>parameters to pass to <a href="#readindata"><code>readindata</code></a></td>
</tr>
</tbody>
</table>
<h3 id="details">Details</h3>
<p>For the <code>taxlevel</code> parameter, each level is made into a separate heatmap. &quot;seq&quot; makes the heatmap with no collapsing of taxonomic levels.</p>
<h3 id="value-2">Value</h3>
<p>Saves heatmaps to outdir.</p>
<h3 id="examples">Examples</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Not run:
<span class="kw">morphheatmap</span>(<span class="dt">mapfile =</span> <span class="st">&quot;mapfile.txt&quot;</span>, <span class="dt">datafile =</span> <span class="st">&quot;OTU_table.txt&quot;</span>, <span class="dt">outdir =</span> <span class="st">&quot;outputs/graphs&quot;</span>, 
    <span class="dt">sampdepth =</span> <span class="dv">25000</span>, <span class="dt">taxlevel =</span> <span class="kw">c</span>(<span class="st">&quot;Family&quot;</span>, <span class="st">&quot;seq&quot;</span>), <span class="dt">tsvfile =</span> <span class="ot">TRUE</span>)
## End(Not run)</code></pre></div>
<h3 id="source-2">Source</h3>
<p><a href="../R/graphs.R">graphs.R</a></p>
<h2 id="pcoaplot"><code>pcoaplot</code></h2>
<p>PCoA plots</p>
<h3 id="description-4">Description</h3>
<p>PCoA plots</p>
<h3 id="usage-3">Usage</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">pcoaplot</span>(mapfile, datafile, outdir, <span class="dt">amp =</span> <span class="ot">NULL</span>, <span class="dt">sampdepth =</span> <span class="ot">NULL</span>,
  <span class="dt">distm =</span> <span class="st">&quot;binomial&quot;</span>, <span class="dt">filter_species =</span> <span class="fl">0.1</span>, <span class="dt">rarefy =</span> <span class="ot">FALSE</span>,
  <span class="dt">colors =</span> <span class="ot">NULL</span>, ...)</code></pre></div>
<h3 id="arguments-3">Arguments</h3>
<table style="width:43%;">
<colgroup>
<col width="19%"></col>
<col width="23%"></col>
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>mapfile</code></td>
<td>full path to map file</td>
</tr>
<tr class="even">
<td><code>datafile</code></td>
<td>full path to input OTU file (biom or see <a href="#readindata">readindata</a> )</td>
</tr>
<tr class="odd">
<td><code>outdir</code></td>
<td>full path to output directory</td>
</tr>
<tr class="even">
<td><code>amp</code></td>
<td>ampvis2 object. may be specified instead of mapfile and datafile</td>
</tr>
<tr class="odd">
<td><code>sampdepth</code></td>
<td>sampling depth</td>
</tr>
<tr class="even">
<td><code>distm</code></td>
<td>distance measure for PCoA. any that are supported by <a href="#ampordinate">amp_ordinate</a> except for unifrac, wunifrac, and none.</td>
</tr>
<tr class="odd">
<td><code>filter_species</code></td>
<td>Remove low abundant OTU's across all samples below this threshold in percent. Setting this to 0 may drastically increase computation time.</td>
</tr>
<tr class="even">
<td><code>rarefy</code></td>
<td>Logical. Rarefy the OTU table if sampdepth is specified.</td>
</tr>
<tr class="odd">
<td><code>colors</code></td>
<td>(Optional) color vector - length equal to number of TreatmentGroups in mapfile</td>
</tr>
<tr class="even">
<td><code>...</code></td>
<td>parameters to pass to <a href="#readindata"><code>readindata</code></a></td>
</tr>
</tbody>
</table>
<h3 id="value-3">Value</h3>
<p>Saves pcoa plots to outdir.</p>
<h3 id="source-3">Source</h3>
<p><a href="../R/graphs.R">graphs.R</a></p>
<h2 id="rarefactioncurve"><code>rarefactioncurve</code></h2>
<p>Make rarefaction curve graph</p>
<h3 id="description-5">Description</h3>
<p>Make rarefaction curve graph</p>
<h3 id="usage-4">Usage</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">rarefactioncurve</span>(mapfile, datafile, outdir, <span class="dt">amp =</span> <span class="ot">NULL</span>, <span class="dt">colors =</span> <span class="ot">NULL</span>, ...)</code></pre></div>
<h3 id="arguments-4">Arguments</h3>
<table style="width:43%;">
<colgroup>
<col width="19%"></col>
<col width="23%"></col>
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>mapfile</code></td>
<td>full path mapping file</td>
</tr>
<tr class="even">
<td><code>datafile</code></td>
<td>ffull path to input OTU file (biom or see <a href="#readindata">readindata</a> )</td>
</tr>
<tr class="odd">
<td><code>outdir</code></td>
<td>full path to output directory</td>
</tr>
<tr class="even">
<td><code>amp</code></td>
<td>(Optional) ampvis2 object. may be specified instead of mapfile and datafile</td>
</tr>
<tr class="odd">
<td><code>colors</code></td>
<td>(Optional) color vector - length equal to number of TreatmentGroups in mapfile</td>
</tr>
<tr class="even">
<td><code>...</code></td>
<td>parameters to pass to <a href="#readindata"><code>readindata</code></a></td>
</tr>
</tbody>
</table>
<h3 id="value-4">Value</h3>
<p>Saves rarefaction curve plot to output directory.</p>
<h3 id="source-4">Source</h3>
<p><a href="../R/graphs.R">graphs.R</a></p>
<h2 id="readindata"><code>readindata</code></h2>
<p>Read in data</p>
<h3 id="description-6">Description</h3>
<p>Read in data</p>
<h3 id="usage-5">Usage</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">readindata</span>(mapfile, datafile, <span class="dt">tsvfile =</span> <span class="ot">FALSE</span>, <span class="dt">mincount =</span> <span class="dv">10</span>)</code></pre></div>
<h3 id="arguments-5">Arguments</h3>
<table style="width:43%;">
<colgroup>
<col width="19%"></col>
<col width="23%"></col>
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>mapfile</code></td>
<td>full path to mapfile. must contain SampleID, TreatmentGroup, and Description columns</td>
</tr>
<tr class="even">
<td><code>datafile</code></td>
<td>full path to input data file. must be either biom file or tab delimited text file. See details.</td>
</tr>
<tr class="odd">
<td><code>tsvfile</code></td>
<td>Logical. Is datafile a tab-delimited text file? See details.</td>
</tr>
<tr class="even">
<td><code>mincount</code></td>
<td>minimum number of reads</td>
</tr>
</tbody>
</table>
<h3 id="details-1">Details</h3>
<p>datafile may be either biom file or text file. If text file, it should have ampvis2 OTU table format <a href="https://madsalbertsen.github.io/ampvis2/reference/amp_load.html#the-otu-table" class="uri">https://madsalbertsen.github.io/ampvis2/reference/amp_load.html#the-otu-table</a> . If the number of reads is less than mincount, the function will give an error, as we cannot make graphs with so few counts.</p>
<h3 id="value-5">Value</h3>
<p>ampvis2 object</p>
<h3 id="source-5">Source</h3>
<p><a href="../R/graphs.R">graphs.R</a></p>
<h2 id="trygraphwrapper"><code>trygraphwrapper</code></h2>
<p>Wrapper for any graph function</p>
<h3 id="description-7">Description</h3>
<p>This is a wrapper for any of the graph functions meant to be called using rpy2 in python.</p>
<h3 id="usage-6">Usage</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">trygraphwrapper</span>(mapfile, datafile, outdir, FUN, <span class="dt">logfilename =</span> <span class="st">&quot;logfile.txt&quot;</span>,
  <span class="dt">info =</span> <span class="ot">TRUE</span>, ...)</code></pre></div>
<h3 id="arguments-6">Arguments</h3>
<table>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>mapfile</code></td>
<td>full path to map file</td>
</tr>
<tr class="even">
<td><code>datafile</code></td>
<td>full path to input OTU file (biom or see <a href="#readindata">readindata</a> )</td>
</tr>
<tr class="odd">
<td><code>outdir</code></td>
<td>output directory for graphs</td>
</tr>
<tr class="even">
<td><code>FUN</code></td>
<td>function you would like to run</td>
</tr>
<tr class="odd">
<td><code>logfilename</code></td>
<td>logfilename</td>
</tr>
<tr class="even">
<td><code>info</code></td>
<td>print sessionInfo to logfile</td>
</tr>
<tr class="odd">
<td><code>...</code></td>
<td>parameters needed to pass to FUN</td>
</tr>
</tbody>
</table>
<h3 id="value-6">Value</h3>
<p>Returns 0 if FUN succeeds and 1 if it returns an error.</p>
<h3 id="examples-1">Examples</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Not run:

<span class="co"># example with no optional arguments for running allgraphs</span>
<span class="kw">trygraphwrapper</span>(<span class="st">&quot;/path/to/inputs/mapfile.txt&quot;</span>, <span class="st">&quot;/path/to/outputs/out.biom&quot;</span>, <span class="st">&quot;/path/to/outputs/&quot;</span>, 
    allgraphs)

<span class="co"># example with optional argument sampdepth</span>
<span class="kw">trygraphwrapper</span>(<span class="st">&quot;/path/to/inputs/mapfile.txt&quot;</span>, <span class="st">&quot;/path/to/outputs/out.biom&quot;</span>, <span class="st">&quot;/path/to/outputs/&quot;</span>, 
    allgraphs, <span class="dt">sampdepth =</span> <span class="dv">30000</span>)

<span class="co"># example of making heatmap with optional arguments</span>
<span class="kw">trygraphwrapper</span>(<span class="st">&quot;/path/to/inputs/mapfile.txt&quot;</span>, <span class="st">&quot;/path/to/outputs/taxa_species.biom&quot;</span>, 
    <span class="st">&quot;/path/to/outputs&quot;</span>, morphheatmap, <span class="dt">sampdepth =</span> <span class="dv">30000</span>, <span class="dt">filter_level =</span> <span class="fl">0.01</span>, <span class="dt">taxlevel =</span> <span class="kw">c</span>(<span class="st">&quot;Family&quot;</span>, 
        <span class="st">&quot;seq&quot;</span>))
## End(Not run)</code></pre></div>
<h3 id="source-6">Source</h3>
<p><a href="../R/graphs.R">graphs.R</a></p>
<h1 id="internal">Internal</h1>
<h2 id="datavis16s-package"><code>datavis16s-package</code></h2>
<p>dataviz16s: A package for Nephele 16S pipeline visualization</p>
<h3 id="description-8">Description</h3>
<p>dataviz16s: A package for Nephele 16S pipeline visualization</p>
<h2 id="filterlowabund"><code>filterlowabund</code></h2>
<p>Filter low abundant taxa</p>
<h3 id="description-9">Description</h3>
<p>Filter low abundant taxa</p>
<h3 id="usage-7">Usage</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">filterlowabund</span>(amp, <span class="dt">level =</span> <span class="fl">0.01</span>, <span class="dt">persamp =</span> <span class="dv">0</span>, <span class="dt">abs =</span> <span class="ot">FALSE</span>)</code></pre></div>
<h3 id="arguments-7">Arguments</h3>
<table style="width:43%;">
<colgroup>
<col width="19%"></col>
<col width="23%"></col>
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>amp</code></td>
<td>ampvis2 object</td>
</tr>
<tr class="even">
<td><code>level</code></td>
<td>level at which to filter</td>
</tr>
<tr class="odd">
<td><code>persamp</code></td>
<td>percent of samples which must have taxa in common</td>
</tr>
<tr class="even">
<td><code>abs</code></td>
<td>is level an absolute count? if false, will use level as relative percent.</td>
</tr>
</tbody>
</table>
<h3 id="value-7">Value</h3>
<p>filtered ampvis2 object</p>
<h3 id="source-7">Source</h3>
<p><a href="../R/utilities.R">utilities.R</a></p>
<h2 id="gridcode"><code>gridCode</code></h2>
<p>Format plotly grid code</p>
<h3 id="usage-8">Usage</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">gridCode</span>(data)</code></pre></div>
<h3 id="arguments-8">Arguments</h3>
<table>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>data</code></td>
<td>data to populate plotly grid</td>
</tr>
</tbody>
</table>
<h3 id="value-8">Value</h3>
<p>list of 3 values:</p>
<ul>
<li><code>jq</code> optional jquery script</li>
<li><code>html</code> html for plotly export link</li>
<li><code>javascript</code> js function for exporting data</li>
</ul>
<h3 id="source-8">Source</h3>
<p><a href="../R/plotlyGrid.R">plotlyGrid.R</a></p>
<h2 id="highertax"><code>highertax</code></h2>
<p>return tables at higher tax level</p>
<h3 id="description-10">Description</h3>
<p>return tables at higher tax level</p>
<h3 id="usage-9">Usage</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">highertax</span>(amp, <span class="dt">taxlevel =</span> <span class="ot">NULL</span>)</code></pre></div>
<h3 id="arguments-9">Arguments</h3>
<table>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>amp</code></td>
<td>ampvis2 object</td>
</tr>
<tr class="even">
<td><code>taxlevel</code></td>
<td>taxonomic level at which to sum up the counts</td>
</tr>
</tbody>
</table>
<h3 id="value-9">Value</h3>
<p>ampvis2 object with otu table and taxa summed up to the taxlevel</p>
<h3 id="source-9">Source</h3>
<p><a href="../R/utilities.R">utilities.R</a></p>
<h2 id="logoutput"><code>logoutput</code></h2>
<p>write log output</p>
<h3 id="description-11">Description</h3>
<p>Prints time along with log message.</p>
<h3 id="usage-10">Usage</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">logoutput</span>(c, <span class="dt">bline =</span> <span class="dv">0</span>, <span class="dt">aline =</span> <span class="dv">0</span>)</code></pre></div>
<h3 id="arguments-10">Arguments</h3>
<table>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>c</code></td>
<td>String. Log message/command to print.</td>
</tr>
<tr class="even">
<td><code>bline</code></td>
<td>Number of blank lines to precede output.</td>
</tr>
<tr class="odd">
<td><code>aline</code></td>
<td>Number of blank lines to follow output.</td>
</tr>
</tbody>
</table>
<h3 id="source-10">Source</h3>
<p><a href="../R/utilities.R">utilities.R</a></p>
<h2 id="plotlygrid"><code>plotlyGrid</code></h2>
<p>Add Plotly data export to Plotly graph</p>
<h3 id="description-12">Description</h3>
<p>All functions create an output html plot with link which sends the data to a grid in the plotly chart studio.</p>
<p><code>plotlyGrid</code> takes in a ggplot or plotly object and creates an output html plotly plot.</p>
<p><code>nonplotlyGrid</code> takes in an htmlwidget.</p>
<p><code>htmlGrid</code> takes in an html tag object.</p>
<h3 id="usage-11">Usage</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotlyGrid</span>(pplot, filename, <span class="dt">data =</span> <span class="ot">NULL</span>, <span class="dt">title =</span> <span class="ot">NULL</span>, <span class="dt">outlib =</span> <span class="st">&quot;lib&quot;</span>)
<span class="kw">nonplotlyGrid</span>(hw, filename, data, <span class="dt">jquery =</span> <span class="ot">FALSE</span>, <span class="dt">title =</span> <span class="ot">NULL</span>,
  <span class="dt">outlib =</span> <span class="st">&quot;lib&quot;</span>)
<span class="kw">htmlGrid</span>(ht, filename, data, <span class="dt">jquery =</span> <span class="ot">FALSE</span>, <span class="dt">title =</span> <span class="ot">NULL</span>, <span class="dt">outlib =</span> <span class="st">&quot;lib&quot;</span>)</code></pre></div>
<h3 id="arguments-11">Arguments</h3>
<table style="width:43%;">
<colgroup>
<col width="19%"></col>
<col width="23%"></col>
</colgroup>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>pplot</code></td>
<td>plotly or ggplot object</td>
</tr>
<tr class="even">
<td><code>filename</code></td>
<td>output filename (fullpath)</td>
</tr>
<tr class="odd">
<td><code>data</code></td>
<td>data frame to export to plotly grid (optional for plotlyGrid)</td>
</tr>
<tr class="even">
<td><code>title</code></td>
<td>title of html page</td>
</tr>
<tr class="odd">
<td><code>outlib</code></td>
<td>(Optional) name of external lib directory for non-selfcontained html. Useful for multiple graphs sharing the same lib.</td>
</tr>
<tr class="even">
<td><code>hw</code></td>
<td>htmlwidget</td>
</tr>
<tr class="odd">
<td><code>jquery</code></td>
<td>should we load jquery</td>
</tr>
<tr class="even">
<td><code>ht</code></td>
<td>html tagList</td>
</tr>
</tbody>
</table>
<h3 id="value-10">Value</h3>
<p>html plot is saved to filename. external libraries are saved to outlib in same directory as filename.</p>
<h3 id="source-11">Source</h3>
<p><a href="../R/plotlyGrid.R">plotlyGrid.R</a></p>
<p><a href="../R/plotlyGrid.R">plotlyGrid.R</a></p>
<p><a href="../R/plotlyGrid.R">plotlyGrid.R</a></p>
<h2 id="print_ampvis2"><code>print_ampvis2</code></h2>
<p>Print ampvis2 object summary</p>
<h3 id="description-13">Description</h3>
<p>Print ampvis2 object summary</p>
<h3 id="usage-12">Usage</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">print_ampvis2</span>(data)</code></pre></div>
<h3 id="arguments-12">Arguments</h3>
<table>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>data</code></td>
<td>ampvis2 object</td>
</tr>
</tbody>
</table>
<h3 id="value-11">Value</h3>
<p>Prints summary stats about ampvis2 object</p>
<h3 id="note">Note</h3>
<p>This is a copy of the internal ampvis2 function print.ampvis2. CRAN does not allow ':::' internal calling of function in package.</p>
<h3 id="source-12">Source</h3>
<p><a href="../R/utilities.R">utilities.R</a></p>
<h2 id="save_fillhtml"><code>save_fillhtml</code></h2>
<p>Save an HTML object to a file</p>
<h3 id="description-14">Description</h3>
<p>Save an HTML object to a file</p>
<h3 id="usage-13">Usage</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">save_fillhtml</span>(html, file, <span class="dt">background =</span> <span class="st">&quot;white&quot;</span>, <span class="dt">libdir =</span> <span class="st">&quot;lib&quot;</span>,
  <span class="dt">bodystyle =</span> <span class="st">&quot;&quot;</span>)</code></pre></div>
<h3 id="arguments-13">Arguments</h3>
<table>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>html</code></td>
<td>HTML content to print</td>
</tr>
<tr class="even">
<td><code>file</code></td>
<td>File to write content to</td>
</tr>
<tr class="odd">
<td><code>background</code></td>
<td>Background color for web page</td>
</tr>
<tr class="even">
<td><code>libdir</code></td>
<td>Directory to copy dependenies to</td>
</tr>
<tr class="odd">
<td><code>bodystyle</code></td>
<td>html style string</td>
</tr>
</tbody>
</table>
<h3 id="value-12">Value</h3>
<p>save html to file</p>
<h3 id="source-13">Source</h3>
<p><a href="../R/plotlyGrid.R">plotlyGrid.R</a></p>
<h2 id="shortnames"><code>shortnames</code></h2>
<p>shortnames for taxonomy</p>
<h3 id="description-15">Description</h3>
<p>shortnames for taxonomy</p>
<h3 id="usage-14">Usage</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">shortnames</span>(taxtable, <span class="dt">taxa =</span> <span class="st">&quot;Species&quot;</span>)</code></pre></div>
<h3 id="arguments-14">Arguments</h3>
<table>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>taxtable</code></td>
<td>taxonomy table object from amp$tax</td>
</tr>
<tr class="even">
<td><code>taxa</code></td>
<td>taxonomic level at which to retrieve the names</td>
</tr>
</tbody>
</table>
<h3 id="value-13">Value</h3>
<p>Vector of shortened names for each seqvar/otu.</p>
<h3 id="source-14">Source</h3>
<p><a href="../R/utilities.R">utilities.R</a></p>
<h2 id="subsetamp"><code>subsetamp</code></h2>
<p>Subset and rarefy OTU table.</p>
<h3 id="description-16">Description</h3>
<p>Subset and/or rarefy OTU table.</p>
<h3 id="usage-15">Usage</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">subsetamp</span>(amp, <span class="dt">sampdepth =</span> <span class="ot">NULL</span>, <span class="dt">rarefy =</span> <span class="ot">FALSE</span>, ...)</code></pre></div>
<h3 id="arguments-15">Arguments</h3>
<table>
<thead>
<tr class="header">
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>amp</code></td>
<td>ampvis2 object</td>
</tr>
<tr class="even">
<td><code>sampdepth</code></td>
<td>sampling depth. See details.</td>
</tr>
<tr class="odd">
<td><code>rarefy</code></td>
<td>rarefy the OTU table in addition to subsetting</td>
</tr>
<tr class="even">
<td><code>...</code></td>
<td>other parameters to pass to amp_subset_samples</td>
</tr>
</tbody>
</table>
<h3 id="details-2">Details</h3>
<p><code>sampdepth</code> will be used to filter out samples with fewer than this number of reads. If rarefy is TRUE, then it will also be used as the depth at which to subsample using vegan function rrarefy.</p>
<h3 id="value-14">Value</h3>
<p>ampvis2 object</p>
<h3 id="source-15">Source</h3>
<p><a href="../R/graphs.R">graphs.R</a></p>

</body>
</html>
